src/animation/mod.rs

```diff
@@ -93,6 +95,7 @@ impl PlayingAnimation2D {
     /// Update the animation given the delta time and the duration of the clip being played.
     #[inline]
     fn update(&mut self, delta: f32, clip_duration: f32) {
+        self.duration = clip_duration;
```

Comment:
The thing is that this is only called if we are not paused or the player has changed. So in the current form paused animation do not hot reload. But that is already the case anyways - I will have to fix that.

Reply:
Ah yeah, I changed that now, should now work regardless of the pause. Though we still have a frame delay of the animation duration, you can see in the example that there is at least one frame where the duration is actually not set, so having the user explicitly handle this case is actually kinda important (I thought this would only be the case at the initialization of the AnimationPlayer2D, but doesn't seem to be the case).